### ISSUE TITLE
i18n support bug

### ISSUE THREAD

**Issue by xxldm** [victorgarciaesgi/nuxt-typed-router#77]
环境
```
strategy: "prefix"
"@nuxtjs/i18n": "8.0.0-beta.10"
"nuxt-typed-router": "^3.0.3"
```
如果语言代码包含`-`，会导致`hasi18nSibling`函数的正则表达式报错。
If the language code contains `-`, it will cause the regular expression of the `hasi18nSibling` function to report an error.
![Y(JTRNONRY410L81OBQF{YE](https://user-images.githubusercontent.com/38677850/222896306-34118461-ad29-4efd-baa5-74fcc61fd206.png)
![%WWJDHPQO}A_VE%`K)EV(KE](https://user-images.githubusercontent.com/38677850/222896310-7cb8a4a9-f7ea-4be1-a444-4ec61608cdbd.png)
并且正则表达式中出现了两个语言代码。这是它们的顺序。
And two language codes appear in the regular expression. Here is their order.
![AM}EB_EMKR7GY{C0%D799IO](https://user-images.githubusercontent.com/38677850/222898785-4037dc72-9ab8-4091-90e5-87a9cf130405.png)

该正则表达式的匹配结果也许会导致问题？
Maybe the result of this regex match is causing the problem?
<img width="472" alt="image" src="https://user-images.githubusercontent.com/38677850/222899183-d723d065-8b89-4d88-b0ca-5f9831592589.png">

**Comment by victorgarciaesgi** [comment]
Thanks for the report!
Yeah I think my regex is bad, I will see how to support `xx-XX` locales codes! 😁

**Comment by victorgarciaesgi** [comment]
Fixed in `3.0.4` !

**Comment by victorgarciaesgi** [comment]
Not completly fixed sorry

**Comment by victorgarciaesgi** [comment]
Now fixed on `3.0.5`
